fastlane_bot/pool_finder.py

```diff
@@ -108,18 +85,17 @@ def _find_pools(self, unsupported_pairs: List[Tuple]) -> Dict[str, List[str]]:
                 mc = MultiCaller(contract=exchange.factory_contract, web3=self._web3, multicall_address=self._multicall_address)
                 for pair in pair_chunk:
                     if exchange.base_exchange_name in [UNISWAP_V2_NAME, SOLIDLY_V2_NAME]:
-                        exchange.get_pool_with_multicall(mc, pair[0], pair[1])
+                        mc.add_call(exchange.get_pool_func_call(pair[0], pair[1]))
```

Comment:
Unfortunately the way `mc.add_call` and `get_pool_func_call` was changed doesn't work. 

`mc.add_call` is expecting the following format for args: 
`mc.add_call(FUNCTION_FROM_CONTRACT, args)`

For example:
`mc.add_call(factory_contract.functions.getPair, tkn0, tkn1, False)`

So to remove the Multicaller dependency, we would need to change the Exchange `get_pool_func_call` to return the function with no args and modify the main loop here - likely also splitting apart Uni V2 & Solidy because they have different inputs. 

If what I wrote doesn't make enough sense please LMK!

Reply:
That's how it works in the new MC (PR #641); I'll update this branch accordingly.